src/Events/Span.php

```diff
@@ -56,18 +56,24 @@ public function __construct(string $name, EventBean $parent)
     {
         parent::__construct([]);
         $this->name  = trim($name);
-        $this->timer = new Timer();
         $this->setParent($parent);
     }
 
     /**
      * Start the Timer
      *
+     * @param float|null $startTime
      * @return void
+     * @throws \Nipwaayoni\Exception\Timer\AlreadyRunningException
      */
-    public function start()
+    public function start(float $startTime = null)
     {
-        $this->timer->start();
+        $this->timer = $this->createTimer($startTime);
+    }
+
+    protected function createTimer(float $startTime = null): Timer
```

Comment:
For now, I'll leave it as `createTimer` since the current mock approach must return a Timer, not set one on `$this`. The check was added to the `start` method so the call to this method is still contextually a `create` act rather than a `get` act.